api/src/test/java/jakarta/data/displaynamegeneration/ReplaceCamelCaseAndUnderscore.java

```diff
@@ -0,0 +1,88 @@
+package jakarta.data.displaynamegeneration;
+
+import org.junit.jupiter.api.DisplayNameGenerator;
+
+import java.lang.reflect.Method;
+
+public class ReplaceCamelCaseAndUnderscore extends DisplayNameGenerator.Standard {
+
+    public ReplaceCamelCaseAndUnderscore() {
+        super();
+    }
+
+    @Override
+    public String generateDisplayNameForClass(Class<?> testClass) {
+        return replaceCamelCaseAndUnderscore(super.generateDisplayNameForClass(testClass));
+    }
+
+    @Override
+    public String generateDisplayNameForNestedClass(Class<?> nestedClass) {
+        return replaceCamelCaseAndUnderscore(super.generateDisplayNameForNestedClass(nestedClass));
+    }
+
+    @Override
+    public String generateDisplayNameForMethod(Class<?> testClass, Method testMethod) {
+        return replaceCamelCaseAndUnderscore(testMethod.getName()) + DisplayNameGenerator.parameterTypesAsString(testMethod);
+    }
+
+    public String replaceCamelCaseAndUnderscore(String input) {
```

Comment:
This method is public for testing purpose but I am sure there is a better way to do that.